firestore/integration_test_internal/src/firestore_test.cc

```diff
@@ -1290,7 +1309,45 @@ TEST_F(FirestoreIntegrationTest, TerminateCanBeCalledMultipleTimes) {
 }
 #endif  // defined(__ANDROID__) && FIRESTORE_HAVE_EXCEPTIONS
 
-TEST_F(FirestoreIntegrationTest, MaintainsPersistenceAfterRestarting) {
+TEST_F(FirestoreTest, CanTerminateFirestoreInstance) {
+  App* app = App::GetInstance();
+  InitResult init_result1;
+  auto db1 =
+      std::unique_ptr<Firestore>(Firestore::GetInstance(app, &init_result1));
+  ASSERT_EQ(kInitResultSuccess, init_result1);
+
+  EXPECT_THAT(db1->Terminate(), FutureSucceeds());
+
+  InitResult init_result2;
+  auto db2 =
+      std::unique_ptr<Firestore>(Firestore::GetInstance(app, &init_result2));
+  ASSERT_EQ(kInitResultSuccess, init_result2);
+
+  EXPECT_NE(db1, db2);
+  db1.reset();
+  db2.reset();
```

Comment:
We need to verify that instance is no longer in cache.

Reply:
TODO: looking for a way to access FirestoreCache from firestore.cc in this file to check that the firestore instances are deleted.

Reply:
remove `reset()`s.